vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/GridFSOutputStreamImpl.java

```diff
@@ -0,0 +1,41 @@
+package io.vertx.ext.mongo.impl;
+
+import com.mongodb.async.SingleResultCallback;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.streams.WriteStream;
+import io.vertx.ext.mongo.GridFSOutputStream;
+
+import java.nio.ByteBuffer;
+
+/**
+ *  Implementation of {@link GridFSOutputStream} which allows streaming with
+ *  Vertx's  {@link io.vertx.core.streams.Pump}
+ *
+ * Adapted from com.github.sth.vertx.mongo.streams.GridFSOutputStreamImpl
+ *
+ * @author <a href="https://github.com/st-h">Steve Hummingbird</a>
+ */
+public class GridFSOutputStreamImpl implements GridFSOutputStream {
+
+    WriteStream<Buffer> writeStream;
+
+    public GridFSOutputStreamImpl(WriteStream<Buffer> writeStream) {
```

Comment:
this class is missing back-pressure handling and error handling

1. before writing to the `WriteStream`, we should check the `writeQueuFull()` method. When it is full, the callback should be delayed and a `drainHandler` should be set on the `WriteStream`. When drain handler is called, the callback should be completed.

2. An `exceptionHandler` should be set on the `WriteStream` and the error should be marked on the `GridFSOutputStreamImpl `. Further calls should check the error status.